### PR TITLE
#6106 SQL init image and using Linux SQL

### PR DESCRIPTION
**Pull request by adoprog**

### CODE REVIEW

File: docker/build/mssql-init/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+FROM ${SPE_ASSETS_IMAGE} AS spe_assets
+FROM ${BASE_IMAGE}
+
+RUN Install-PackageProvider -Name NuGet -Force; `
+    Set-PackageSource -Name PSGallery -Trusted ; `
+    Install-Module -Name SqlServer;
+
+COPY ./sql /sql
+COPY ./tools/scripts\ .\
+COPY ./data /data
```
Comment (jflheureux) [inline docker/build/mssql-init/Dockerfile:17]: Should the `COPY` order be optimized for caching with:
1. Modules (Headless, SPE) first
2. sql
3. tools/scripts
4. data last

File: docker/build/mssql-init/sql/ResetDemoUsers.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
+USE [Sitecore.Core]
+
+DECLARE @UserId uniqueidentifier, @UserName nvarchar(256)
+
+DECLARE uc CURSOR FOR
+SELECT UserId, UserName
+FROM [aspnet_Users]
+WHERE username in ('sitecore\minnie', 'sitecore\pete', 'sitecore\demoadmin', 'sitecore\Content Author', 'sitecore\Marketer')
```
Comment (jflheureux) [inline docker/build/mssql-init/sql/ResetDemoUsers.sql:8]: There are some users we do not have in this demo yet that are listed here. Should they be removed for now?

**Review by jflheureux — CHANGES_REQUESTED** [review]
It should work. I will test once you change it to use mssql 2019.

File: docker-compose.override.yml [hunk 3]
```diff
@@ -64,16 +64,38 @@ services:
   # module data added. See Dockerfile for details.
   mssql:
     image: ${REGISTRY}${COMPOSE_PROJECT_NAME}-xm1-mssql:${VERSION:-latest}
+    healthcheck:
+      test: ["CMD", "dir"]
+    image: mcr.microsoft.com/mssql/server:2017-latest
+    mem_limit: 2GB
```
[the remark this answers is not in the document]
Comment (adoprog) [reply docker-compose.override.yml:69]: SQL requires a container with at least 2000 megabytes of memory.

File: docker-compose.override.yml [hunk 4]
```diff
@@ -64,16 +64,38 @@ services:
...
+    volumes:
+      - type: bind
+        source: .\docker\data\sql
+        target: /var/opt/mssql/data
```
[the remark this answers is not in the document]
Comment (adoprog) [reply docker-compose.override.yml:73]: Added

File: docker-compose.override.yml [hunk 5]
```diff
@@ -96,10 +118,14 @@ services:
...
   id:
     image: ${REGISTRY}${COMPOSE_PROJECT_NAME}-id:${VERSION:-latest}
+    depends_on:
+      mssql-init:
+        condition: service_healthy
     build:
       context: ./docker/build/id
       args:
         PARENT_IMAGE: ${SITECORE_DOCKER_REGISTRY}sitecore-id:${SITECORE_VERSION}
+        TOOLS_ASSETS: ${TOOLS_IMAGE}
```
[the remark this answers is not in the document]
Comment (adoprog) [reply docker-compose.override.yml]: Done

File: docker/build/mssql-init/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+COPY --from=headless_services /module/db /jss_data
+COPY --from=spe_assets /module/db /spe_data
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
```
[the remark this answers is not in the document]
Comment (adoprog) [reply docker/build/mssql-init/Dockerfile]: Done

File: docker/build/mssql-init/Dockerfile (hunk 1 above)
Comment (adoprog) [reply docker/build/mssql-init/Dockerfile]: Done

**Pull request merged**